pygatt/bled112_backend.py

```diff
@@ -88,105 +83,118 @@ def __init__(self, serial_port, run=True, logfile=None):
         # Initialization
         self._lib = bled112_bglib.BGLib(loghandler=handler,
                                         loglevel=LOG_LEVEL)
-        # Note: _ser is not protected by _main_thread_cond
         self._ser = serial.Serial(serial_port, timeout=0.25)
 
-        # Main thread (the one calling commands) waits on this. This also
-        # provides mutual exclustion for BLED112Backend's state (except for the
-        # log)
-        self._main_thread_cond = threading.Condition()
-
-        # Packet receiving
-        self._recvr_thread = None  # background thread to receive packets
-        self._recvr_stop = True  # tell recvr thread not to run
-
-        # Flags to tell the main thread what to do when woken up from waiting
-        self._attribute_value_received = False  # attribute_value event occurred
-        self._bonded = False  # device is bonded
-        self._bond_expected = False  # tell bond_status handler to set _bonded
-        self._bonding_fail = False  # failed to bond with device
-        self._connect_timeout = False  # connection procedure timed-out
-        self._connected = False  # device is connected
-        self._encrypted = False  # connection is encrypted
-        self._event_return = 0  # event return code
-        self._procedure_completed = False  # procecure_completed event occurred
-        self._response_received = False  # command response received
-        self._response_return = 0  # command response return code
-
-        # BLED112_backend's state
-        self._attribute_value = None  # attribute_value event value
-        self._expected_attribute_handle = None  # expected handle after a read
+        # State
+        self._num_bonds = 0  # number of bonds stored on the BLED112
+        self._stored_bonds = []  # bond handles stored on the BLED112
         self._bond_handle = 0xFF  # handle for the device bond
```

Comment:
Okay, let's go ahead and remove it.  It's tempting to leave things like this in the code in case you decide to support the feature later, but it's one more thing you need to keep in your cognitive space, which imho isn't worth it.